webpack/assets/javascripts/react_app/components/ColumnSelector/ColumnSelector.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState } from 'react';
+import { PropTypes } from 'prop-types';
+import { Button, Modal, ModalVariant, TreeView } from '@patternfly/react-core';
+import { translate as __ } from '../../common/I18n';
+import API from '../../API';
+import { changeQuery } from '../../common/urlHelpers';
+import './column-selector.scss';
+
+const ColumnSelector = props => {
+  const {
+    data: { url, columns },
+  } = props;
+
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [selectedColumns, setSelectedColumns] = useState(columns);
+
+  const getColumnKeys = event => {
+    return columns
+      .find(category => category.key === 'general')
+      .children.map(col => {
+        if (col.checkProps.checked) {
+          return col.key;
+        }
+        return null;
+      })
+      .filter(item => item);
+  };
+
+  const updateTablePreference = event => {
+    API.put(url, { columns: getColumnKeys() });
+    changeQuery({});
+  };
+
+  const onClose = event => {
```

Comment:
If users don't confirm their changes the selector behaves oddly. If I check/uncheck a column and simply close the form, and re-open it again, I see my changes are applied to the selector, but the table is unchanged. What I mean is that I think we better return the state of the selector before any changes since they weren't applied.